File: server/model/resource_spec.go
package model

import (
	"time"
)

type ResourceSpec struct {
	ID        uint      `json:"id" gorm:"primaryKey"`
	Name      string    `json:"name" gorm:"type:varchar(100);uniqueIndex;not null"`
	CPUCores  int       `json:"cpu_cores" gorm:"not null"`
	MemoryGB  int       `json:"memory_gb" gorm:"not null"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

func (ResourceSpec) TableName() string {
	return "resource_specs"
}

type CreateResourceSpecRequest struct {
	Name     string `json:"name" binding:"required"`
	CPUCores int    `json:"cpu_cores" binding:"required"`
	MemoryGB int    `json:"memory_gb" binding:"required"`
}

type UpdateResourceSpecRequest struct {
	Name     string `json:"name" binding:"required"`
	CPUCores int    `json:"cpu_cores" binding:"required"`
	MemoryGB int    `json:"memory_gb" binding:"required"`
}

type ResourceSpecListResponse struct {
	List     []ResourceSpec `json:"list"`
	Total    int64          `json:"total"`
	Page     int            `json:"page"`
	PageSize int            `json:"page_size"`
}

File: web/src/api/resourceSpec.ts
/**
 * 资源规格管理 API
 * 对应后端 /resource-specs 路由组（管理员可写，所有已认证用户可读列表）
 */
import service from './client'
import type { ApiResponse } from '@/types/api'

/** 资源规格项（与后端 model.ResourceSpec 对齐） */
export interface ResourceSpecItem {
  id: number
  name: string
  /** CPU 核心数 */
  cpu_cores: number
  /** 内存大小（GB） */
  memory_gb: number
  created_at: string
  updated_at: string
}

/** 列表响应（与后端 ResourceSpecListResponse 对齐） */
export interface ResourceSpecListResult {
  list: ResourceSpecItem[]
  total: number
  page: number
  page_size: number
}

/** 创建/更新请求体 */
export interface ResourceSpecPayload {
  name: string
  cpu_cores: number
  memory_gb: number
}

/** 获取资源规格列表（支持分页与关键字搜索） */
export function listResourceSpecs(params?: { page?: number; page_size?: number; keyword?: string }) {
  return service.get<unknown, ApiResponse<ResourceSpecListResult>>('/resource-specs', {
    params,
    silent: true,
  })
}

/** 创建资源规格 */
export function createResourceSpec(data: ResourceSpecPayload) {
  return service.post<unknown, ApiResponse<ResourceSpecItem>>('/resource-specs', data)
}

/** 更新资源规格 */
export function updateResourceSpec(id: number, data: ResourceSpecPayload) {
  return service.put<unknown, ApiResponse<ResourceSpecItem>>(`/resource-specs/${id}`, data)
}

/** 删除资源规格 */
export function deleteResourceSpec(id: number) {
  return service.delete<unknown, ApiResponse>(`/resource-specs/${id}`)
}

/** 批量删除资源规格 */
export function batchDeleteResourceSpecs(ids: number[]) {
  return service.post<unknown, ApiResponse>('/resource-specs/batch-delete', { ids })
}
